src/google-maps/map-info-window/map-info-window.ts

```diff
@@ -0,0 +1,184 @@
+  private readonly _position = new BehaviorSubject<google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _content = new ReplaySubject<Node>(1);
+
+  private readonly _listeners: google.maps.MapsEventListener[] = [];
+
+  private readonly _destroy = new Subject<void>();
+
+  private _infoWindow?: google.maps.InfoWindow;
+
+  constructor(private readonly googleMap: GoogleMap) {}
+
+  ngOnInit() {
+    this._combineOptions().pipe(takeUntil(this._destroy)).subscribe(options => {
+      if (this._infoWindow) {
+        this._infoWindow.setOptions(options);
+      } else {
+        this._infoWindow = new google.maps.InfoWindow(options);
+        this._initializeEventHandlers();
+      }
+    });
+  }
+
+  ngOnDestroy() {
+    this._destroy.next();
+    this._destroy.complete();
+    for (let listener of this._listeners) {
+      listener.remove();
+    }
+    this.close();
+  }
+
+  /**
+   * See developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.close
+   */
+  close() {
+    if (this._infoWindow) {
+      this._infoWindow.close();
+    }
+  }
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.getContent
+   */
+  getContent(): string|Node {
+    return this._infoWindow!.getContent();
+  }
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/info-window
+   * #InfoWindow.getPosition
+   */
+  getPosition(): google.maps.LatLng|null {
+    return this._infoWindow!.getPosition() || null;
+  }
+
+  /**
+   * See
+   * developers.google.com/maps/documentation/javascript/reference/info-window#InfoWindow.getZIndex
+   */
+  getZIndex(): number {
+    return this._infoWindow!.getZIndex();
+  }
+
+  /**
+   * Opens the MapInfoWindow using the provided MapMarker as the anchor. If the anchor is not set,
+   * then the position property of the options input is used instead.
+   */
+  open(anchor?: MapMarker) {
+    const marker = anchor ? anchor._marker : undefined;
+    if (this.googleMap._googleMap) {
+      this._infoWindow!.open(this.googleMap._googleMap, marker);
+    }
+  }
+
+  private _combineOptions(): Observable<google.maps.InfoWindowOptions> {
+    return combineLatest(this._options, this._position, this._content)
+        .pipe(map(([options, position, content]) => {
+          const combinedOptions: google.maps.InfoWindowOptions = {
+            ...options,
+            position: position || options.position,
+            content,
+          };
+          return combinedOptions;
+        }));
+  }
+
+  private _initializeEventHandlers() {
+    const eventHandlers = new Map<string, EventEmitter<void>>([
+      ['closeclick', this.closeclick],
+      ['content_changed', this.contentChanged],
+      ['domready', this.domready],
+      ['position_changed', this.positionChanged],
+      ['zindex_changed', this.zindexChanged],
+    ]);
+    eventHandlers.forEach((eventHandler: EventEmitter<void>, name: string) => {
```

Comment:
For a follow-up: this could probably be moved to a common place to be shared among the maps components. 

Now that I think about it, too, I would also forgo the `Map` data structure since you don't actually need to ever look up values, so just iterating over an array of tuples would be be doing less work

Reply:
Ack. Will do.